### PR TITLE
Add globally blocked dnsname infrastructure

### PR DESCRIPTION
**Pull request by furkansahin**
**Add globally blocked dnsname infrastructure migration**

**Add globally blocked dnsnames to firewall rules** 
The nftables that is used for firewall rules gets 2 new sets. These are
used to block incoming and outgoing traffic to select IP addresses.

**Add prog to update ip addresses of globally blocked dns names**
This commit introduces a new prog that will periodically check the ip
addresses of globally blocked dnsnames. This way, the list will be kept
fresh and the new VMs will be provisioned with the new list. In future,
we might introduce a system here to trigger a firewall rule update for
existing VMs as well.

### CODE REVIEW

**Comment by fdr** [comment]
maybe it does something, it depends how adversarial the dns recrods are, e.g. if they implement [fast flux dns](https://en.wikipedia.org/wiki/Fast_flux). We can probably review that on our own with a few DNS queries over a couple of days.

If it is fluxed, we may need to go to various databases or other approaches to deter this violation of TOS.

File: prog/vnet/update_firewall_rules.rb [hunk 1]
```diff
@@ -88,6 +106,22 @@ class Prog::Vnet::UpdateFirewallRules < Prog::Base
     pop "firewall rule is added"
   end
 
+  def generate_globally_blocked_lists
+    globally_blocked_ipv4s = []
+    globally_blocked_ipv6s = []
+
+    GloballyBlockedDnsname.all.each do |globally_blocked_dnsname|
```
Comment (fdr) [inline prog/vnet/update_firewall_rules.rb:113]: all.each again: necessary? Or just each?

**Review by fdr — APPROVED** [review]
Minor matters, probably the most interesting general design choice was to mix ip4 and ip6 together and then separate them again as they come out of DNS separately (A, AAAA). But not necessarily a bad choice.

File: prog/resolve_globally_blocked_dnsnames.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require "socket"
+require "open-uri"
+require "net/http"
+class Prog::ResolveGloballyBlockedDnsnames < Prog::Base
+  label def wait
+    current_frame = strand.stack.first
```
[the remark this answers is not in the document]
Comment (furkansahin) [reply prog/resolve_globally_blocked_dnsnames.rb]: Yes, this fails with freezing. Though, upon further looking into this. I decided we do not need it anyway.

**Pull request merged**